DigiMeSDK/Core/Classes/Security/DMEJWTUtility.swift

```diff
@@ -0,0 +1,446 @@
+            case preAuthCode = "preauthorization_code"
+        }
+    }
+    
+    // claims to request a authorization code
+    class PayloadRequestAuthJWT: NSObject, Decodable, Claims {
+        var clientId: String?
+        var code: String?
+        var codeVerifier: String?
+        var grantType: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var timestamp: Double?
+
+        enum CodingKeys: String, CodingKey {
+            case clientId = "client_id"
+            case code
+            case codeVerifier = "code_verifier"
+            case grantType = "grant_type"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case timestamp
+        }
+    }
+    
+    // claims to validate authorization and refresh tokens
+    class PayloadValidateAuthJWT: NSObject, Decodable, Claims {
+        var accessToken: String?
+        var expiresTimestamp: Double?
+        var refreshToken: String?
+        var tokenType: String?
+        
+        enum CodingKeys: String, CodingKey {
+            case accessToken = "access_token"
+            case expiresTimestamp = "expires_on"
+            case refreshToken = "refresh_token"
+            case tokenType = "token_type"
+        }
+    }
+    
+    // claims to request data trigger
+    class PayloadDataTriggerJWT: NSObject, Decodable, Claims {
+        var accessToken: String?
+        var clientId: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var sessionKey: String?
+        var timestamp: Double?
+
+        enum CodingKeys: String, CodingKey {
+            case accessToken = "access_token"
+            case clientId = "client_id"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case sessionKey = "session_key"
+            case timestamp
+        }
+    }
+    
+    // claims to request OAuth token renewal
+    class PayloadRefreshOAuthJWT: NSObject, Decodable, Claims {
+        var clientId: String?
+        var grantType: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var refreshToken: String?
+        var timestamp: Double?
+        
+        enum CodingKeys: String, CodingKey {
+            case clientId = "client_id"
+            case grantType = "grant_type"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case refreshToken = "refresh_token"
+            case timestamp
+        }
+    }
+
+    /// Creates request JWT which can be used to get a preAuthentication token
+    ///
+    /// - Parameters:
+    ///   - appId: application identifier
+    ///   - contractId: contract identifier
+    ///   - privateKey: private key in base 64 format
+    ///   - publicKey: public key in base 64 format
+    @objc public class func signedPreAuthJwt(_ appId: String, contractId: String, privateKey: String, publicKey: String?) -> String? {
+        guard
+            privateKey.isHex(),
```

Comment:
Ah hang on now, this highlights another issue. We say we expect keys in `Base64`, but first thing we do is check if the key is in `Hex` format! Should the expectation be changed in description?